### PR TITLE
Add support for composable matchers

### PR DESCRIPTION
**Pull request by Darhazer**
Describe the changes and motivations for the pull request.

### Review

- [ ] Document code according to [Getting Started with Yard](http://www.rubydoc.info/gems/yard/file/docs/GettingStarted.md).
- [ ] All tests are passing.
- [ ] Test manually.
- [ ] Get approval.

### Pre-merge checklist

- [ ] The PR relates to a single subject with a clear title and description in grammatically correct, complete sentences.
- [ ] Verify that feature branch is up-to-date with `master` (if not - rebase it).
- [ ] Double check the quality of [commit messages](http://chris.beams.io/posts/git-commit/).
- [ ] Squash related commits together.

### CODE REVIEW

File: lib/granite/rspec/satisfy_preconditions.rb [hunk 1]
```diff
@@ -94,10 +94,12 @@ def expected_kind_of_messages_error(object, expected_kind_of_messages, message =
...
 
   def compare(error, expected)
-    if expected.is_a? String
+    if RSpec::Matchers.is_a_matcher?(expected)
+      expected.matches?(error)
+    elsif expected.is_a?(String)
       error == expected
     else
-      error.match? expected
+      error.match?(expected)
```
Comment (pirj) [inline lib/granite/rspec/satisfy_preconditions.rb]: Wondering if RSpec's built-in `values_match?` does the same thing as this method.

Comment (pirj) [inline lib/granite/rspec/satisfy_preconditions.rb]: I propose to change `error` to `error_message` or just `message`.
Or, even better, to `actual`.

Comment (Darhazer) [reply lib/granite/rspec/satisfy_preconditions.rb]: This is the first thing I've tried, but tests failed. Didn't dig deeper though

Comment (Darhazer) [reply lib/granite/rspec/satisfy_preconditions.rb]: Updated

Comment (pirj) [reply lib/granite/rspec/satisfy_preconditions.rb]: Looks good 👍

**Pull request merged**